packages/waku/src/router/create-pages.ts

```diff
@@ -457,3 +471,357 @@ export function createPages<
+      if (wildcardPagePathMap.has(page.path)) {
+        throw new Error(`Duplicated dynamic path: ${page.path}`);
+      }
+      wildcardPagePathMap.set(page.path, [pathSpec, page.component]);
+    } else {
+      throw new Error('Invalid page configuration');
+    }
+    return page as Exclude<typeof page, { path: never } | { render: never }>;
+  };
+
+  const createLayout: CreateLayout = (layout) => {
+    if (configured) {
+      throw new Error('createLayout no longer available');
+    }
+    if (layout.render === 'static') {
+      const id = joinPath(layout.path, 'layout').replace(/^\//, '');
+      registerStaticComponent(id, layout.component);
+    } else if (layout.render === 'dynamic') {
+      if (dynamicLayoutPathMap.has(layout.path)) {
+        throw new Error(`Duplicated dynamic path: ${layout.path}`);
+      }
+      const pathSpec = parsePathWithSlug(layout.path);
+      dynamicLayoutPathMap.set(layout.path, [pathSpec, layout.component]);
+    } else {
+      throw new Error('Invalid layout configuration');
+    }
+  };
+
+  const createRoot: CreateRoot = (root) => {
+    if (configured) {
+      throw new Error('createRoot no longer available');
+    }
+    if (rootItem) {
+      throw new Error(`Duplicated root component`);
+    }
+    if (root.render === 'static' || root.render === 'dynamic') {
+      rootItem = root;
+    } else {
+      throw new Error('Invalid root configuration');
+    }
+  };
+
+  let ready: Promise<AllPages | void> | undefined;
+  const configure = async () => {
+    if (!configured && !ready) {
+      ready = fn({ createPage, createLayout, createRoot });
+      await ready;
+      configured = true;
+    }
+    await ready;
+  };
+
+  const getLayouts = (path: string): string[] => {
+    const pathSegments = path.split('/').reduce<string[]>(
+      (acc, segment, index) => {
+        if (segment === '') {
+          return acc;
+        }
+        acc.push(acc[index - 1] + '/' + segment);
+        return acc;
+      },
+      ['/'],
+    );
+
+    return pathSegments.filter(
+      (segment) =>
+        dynamicLayoutPathMap.has(segment) ||
+        staticComponentMap.has(joinPath(segment, 'layout').slice(1)), // feels like a hack
+    );
+  };
+
+  const definedRouter = new_defineRouter({
+    getPathConfig: async () => {
+      await configure();
+      const paths: {
+        pattern: string;
+        path: PathSpec;
+        components: Record<string, { isStatic: boolean }>;
+        noSsr: boolean;
+      }[] = [];
+      const rootIsStatic = !rootItem || rootItem.render === 'static';
+
+      for (const [path, pathSpec] of fixedPathSet) {
+        const noSsr = noSsrSet.has(pathSpec);
+        const isStatic = (() => {
+          for (const [_, [layoutPathSpec]] of dynamicLayoutPathMap) {
+            if (hasPathSpecPrefix(layoutPathSpec, pathSpec)) {
+              return false;
+            }
+          }
+          return true;
+        })();
```

Comment:
I personally prefer array.some to iife.

What does this `isStatic` represent? Doesn't it include checking dynamicPagePathMap?

Reply:
the problem is that the dynamic layout path map stores the path spec in the values. Maybe this is a sign that this structure in particular could be redone to make the pathspec the key of the map, so that we can just do `dynamicLayoutPathMap.has(pathSpec)`

Reply:
actually this is being used: https://github.com/tylersayshi/waku/blob/new-create-pages/packages/waku/src/router/create-pages.ts#L324

I am good with changing to `array.some`, but I am going to do some more thinking about "What does this isStatic represent?" - I'd like to have a better answer and mostly carried this over from the original iteration of `createPages` without thinking much about it

Reply:
removed the iife